Tasks/AzureNLBManagement/nlbazureutility.ts

```diff
@@ -0,0 +1,215 @@
+import * as tl from 'vsts-task-lib/task';
+import * as Q from 'q';
+import * as querystring from "querystring";
+import * as httpClient from 'vso-node-api/HttpClient';
+import * as restClient from 'vso-node-api/RestClient';
+
+var httpObj = new httpClient.HttpCallbackClient(tl.getVariable("AZURE_HTTP_USER_AGENT"));
+var authUrl = 'https://login.windows.net/';
+var azureApiVersion = '2016-09-01';
+	
+function getAccessToken(SPN, endpointUrl: string): Q.Promise<string> {
+
+	var deferred = Q.defer<string>();
+	var authorityUrl = authUrl + SPN.tenantID + "/oauth2/token/";
+
+	var post_data = querystring.stringify({
+		resource: endpointUrl, 
+		client_id: SPN.servicePrincipalClientID,
+		grant_type: "client_credentials", 
+		client_secret: SPN.servicePrincipalKey
+	});
+
+	var requestHeader = {
+		"Content-Type": "application/x-www-form-urlencoded; charset=utf-8"
+	};
+
+	tl.debug(`Requesting for bearer token ${authorityUrl}`);
+	httpObj.send("POST", authorityUrl, post_data, requestHeader, (error, response, body) => {
+		if(error) {
+			deferred.reject(error);
+		}
+		else if (response.statusCode == 200) {
+			deferred.resolve(JSON.parse(body).access_token);
+		}
+		else {
+			deferred.reject(tl.loc("CouldNotFetchAcessToken", response.statusCode, response.statusMessage, body));
+		};
+	}); 
+
+    return deferred.promise;
+}
+
+export async function getNetworkInterfacesInRG(SPN, endpointUrl: string, resourceGroupName: string) {
+
+	var deferred = Q.defer<any>();
+	var restUrl = "https://management.azure.com/subscriptions/" + SPN.subscriptionId + "/resourceGroups/" + resourceGroupName + "/providers/Microsoft.Network/networkInterfaces?api-version=" + azureApiVersion;
+	var accessToken = await getAccessToken(SPN, endpointUrl);
+
+	var requestHeader = {
+		Authorization: 'Bearer ' + accessToken
+	};
+
+	tl.debug(`Getting all network interfaces in the resource group ${resourceGroupName}`);
+	httpObj.get('GET', restUrl, requestHeader, (error, response, body) => {
+		if(error) {
+			deferred.reject(error);
+		}
+		else if(response.statusCode == 200) {
+			deferred.resolve(JSON.parse(body).value);
+		}
+		else {
+			deferred.reject(tl.loc("CouldNotFetchNetworkInterfacesInRg", resourceGroupName, response.statusCode, response.statusMessage, body));
+		}
+
+	});
+	return deferred.promise;
+}
+
+export async function getLoadBalancer(SPN, endpointUrl: string, name: string, resourceGroupName: string) {
+	
+	var deferred = Q.defer<any>();    
+	var restUrl = "https://management.azure.com/subscriptions/" + SPN.subscriptionId + "/resourceGroups/" + resourceGroupName + "/providers/Microsoft.Network/loadBalancers/" + name + "?api-version=" + azureApiVersion;
+	var accessToken = await getAccessToken(SPN, endpointUrl);
+
+	var requestHeader = {
+		authorization: 'Bearer ' + accessToken
```

Comment:
Since you are bypassing REST Client and using HTTP client, you are missing out filling the Accept header. This comment is applicable to all other places as well. If the issue of using REST client is only in SetNetworkInterface, why can't you use REST client in other places?